javascript/webhooks-nodejs/createWebhook.js

```diff
@@ -0,0 +1,102 @@
+  let envContent = fs.readFileSync(envFilePath, "utf8");
+  envContent = envContent.replace(
+    /X_HOOK_SECRET=.*/,
+    `X_HOOK_SECRET=${newSecret}`
+  );
+  fs.writeFileSync(envFilePath, envContent);
+  console.log(`The X-Hook-Secret stored in .env is: ${newSecret}`);
+}
+
+// Helper function to read X-Hook-Secret from the .env file
+function getXHookSecret() {
+  const envContent = fs.readFileSync(envFilePath, "utf8");
+  const match = envContent.match(/X_HOOK_SECRET=(.*)/);
+  return match ? match[1] : "";
+}
+
+// Async function to delete the webhook
+// Documentation: https://developers.asana.com/reference/deletewebhook
+async function deleteWebhook(webhookId, personalAccessToken) {
+  const url = `https://app.asana.com/api/1.0/webhooks/${webhookId}`;
+
+  try {
+    await axios.delete(url, {
+      headers: {
+        Authorization: `Bearer ${personalAccessToken}`,
+      },
+    });
+    console.log(`Webhook ${webhookId} deleted due to mismatch.`);
+  } catch (error) {
+    console.error(
+      "Error deleting webhook:",
+      error.response ? error.response.data : error.message
+    );
+  }
+}
+
+// Async function to create the webhook
+// Documentation: https://developers.asana.com/reference/createwebhook
+async function createWebhook(targetUri, objectId, filter, resourceType) {
+  const personalAccessToken = process.env.PAT;
+  const url = "https://app.asana.com/api/1.0/webhooks";
+
+  try {
+    const response = await axios.post(
+      url,
+      {
+        data: {
+          resource: objectId,
+          target: targetUri,
+          filters: [{ action: filter, resource_type: resourceType }],
+        },
+      },
+      {
+        headers: {
+          Authorization: `Bearer ${personalAccessToken}`,
+        },
+      }
+    );
+
+    const xHookSecret = response.data["X-Hook-Secret"];
+    const storedSecret = getXHookSecret();
+
+    // Compare the secrets
+    // Read more about validation during the webhook handshake: https://developers.asana.com/docs/webhooks-guide#the-webhook-handshake
+    if (xHookSecret !== storedSecret) {
+      console.error(
+        `X-Hook-Secrets do not match! Potential security issue. Deleting webhook with ID ${response.data.data.gid}`
+      );
+      // Delete the webhook if the secrets don't match
+      await deleteWebhook(response.data.data.gid, personalAccessToken);
+      return;
+    }
+
+    updateXHookSecret(xHookSecret);
+    console.log(
+      `Webhook created successfully! The X-Hook-Secret from Asana's 201 response was: ${xHookSecret}`
+    );
+  } catch (error) {
+    console.error(
+      "Error creating webhook:",
+      error.response ? error.response.data : error.message
+    );
+  }
+}
+
+// TODO: Replace these values with your target URI, object ID, filter, and resource type
+const targetUri = "https://your.url.here/receiveWebhook"; // The webhook server's public endpoint for receiving webhooks
+const objectId = "TASK_ID_HERE"; // The Asana object ID you want to track
```

Comment:
`// The Asana object ID you want to track` and `TASK_ID_HERE` is conflicting maybe add something like `(e.g., task gid)`

Reply:
Fixed!